nav2_rrtconnect_planner/src/rrt_connect.cpp

```diff
@@ -0,0 +1,269 @@
+    bounds_->high[0], bounds_->high[1]);
+
+  ompl_state_space_ = std::make_shared<ompl::base::SE2StateSpace>();
+  // Bounding the R2 state space
+  ompl_state_space_->as<ompl::base::SE2StateSpace>()->setBounds(*bounds_);
+
+  // Simple setup
+  ss_.reset(new ompl::geometric::SimpleSetup(ompl_state_space_));
+  // Setting state validity checker
+  ss_->setStateValidityChecker(
+    [this](const ompl::base::State * state) {
+      return this->isStateValid(state);
+    });
+  // Setting the state validity checking resolution
+  ss_->getSpaceInformation()->setStateValidityCheckingResolution(collision_checking_resolution_);
+
+  ompl_initialized_ = true;
+}
+
+nav_msgs::msg::Path RRTConnect::createPlan(
+  const geometry_msgs::msg::PoseStamped & start,
+  const geometry_msgs::msg::PoseStamped & goal)
+{
+  if (!ompl_initialized_) {
+    RCLCPP_INFO(node_->get_logger(), "Initializing OMPL!");
+    initializeOMPL();
+  }
+  // Returning variable
+  nav_msgs::msg::Path global_path;
+
+  // Checking if the goal and start state is in the global frame
+  if (start.header.frame_id != global_frame_) {
+    RCLCPP_ERROR(
+      node_->get_logger(), "Planner will only except start position from %s frame",
+      global_frame_.c_str());
+    return global_path;
+  }
+
+  if (goal.header.frame_id != global_frame_) {
+    RCLCPP_INFO(
+      node_->get_logger(), "Planner will only except goal position from %s frame",
+      global_frame_.c_str());
+    return global_path;
+  }
+
+  ompl::base::ScopedState<> ss_start(ompl_state_space_), ss_goal(ompl_state_space_);
+  ss_start = std::vector<double>{start.pose.position.x, start.pose.position.y};
+  ss_goal = std::vector<double>{goal.pose.position.x, goal.pose.position.y};
+  // Setting state and goal state
+  ss_->setStartAndGoalStates(ss_start, ss_goal);
+  // Assigning RRTConnect
+  ss_->setPlanner(
+    ompl::base::PlannerPtr(
+      new ompl::geometric::RRTConnect(
+        ss_->getSpaceInformation())));
+  // Load settings
+  ss_->setup();
+  // Counter for replanning
+  int counter = 0;
+  // Planning
+  while (!ss_->solve(solve_time_) && counter < max_trials_) {
+    ++counter;
+  }
+  // Plan found
+  if (counter < max_trials_) {
+    RCLCPP_INFO(
+      node_->get_logger(), "Found %d solutions",
+      ss_->getProblemDefinition()->getSolutionCount());
+    // Path simplification
+    ss_->simplifySolution();
+    auto solution_path = ss_->getSolutionPath();
+
+    global_path.poses.clear();
+    global_path.header.stamp = node_->now();
+    global_path.header.frame_id = global_frame_;
+    // Increasing number of path points
+    int min_num_states = round(solution_path.length() / costmap_->getResolution());
+    solution_path.interpolate(min_num_states);
+
+    global_path.poses.reserve(solution_path.getStates().size());
+    for (const auto ptr : solution_path.getStates()) {
+      global_path.poses.emplace_back(getPoseStampedFromState(*ptr));
```

Comment:
If you're constructing the object, you don't get a speed up on emplace back.

Reply:
Got it!